WordPressKitTests/Utilities/HTTPRequestBuilderTests.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import XCTest
+
+@testable import WordPressKit
+
+class HTTPRequestBuilderTests: XCTestCase {
+
+    func testDefaultMethod() throws {
+        let request = try HTTPRequestBuilder(url: URL(string: "https://wordpress.org")!).build()
+        XCTAssertEqual(request.url?.absoluteString, "https://wordpress.org")
+        XCTAssertEqual(request.httpMethod, "GET")
+    }
+
+    func testHeader() throws {
+        let request = try HTTPRequestBuilder(url: URL(string: "https://wordpress.org")!)
+            .set(value: "Foo", forHeader: "X-Header-1")
+            .set(value: "Bar", forHeader: "X-Header-2")
+            .build()
+        XCTAssertEqual(request.value(forHTTPHeaderField: "X-Header-1"), "Foo")
+        XCTAssertEqual(request.value(forHTTPHeaderField: "X-Header-2"), "Bar")
+    }
+
+    func testPath() throws {
+        var request = try HTTPRequestBuilder(url: URL(string: "https://wordpress.org")!)
+            .set(path: "hello/world")
+            .build()
+        XCTAssertEqual(request.url?.absoluteString, "https://wordpress.org/hello/world")
+
+        request = try HTTPRequestBuilder(url: URL(string: "https://wordpress.org")!)
+            .set(path: "/hello/world")
+            .build()
+        XCTAssertEqual(request.url?.absoluteString, "https://wordpress.org/hello/world")
+
+        request = try HTTPRequestBuilder(url: URL(string: "https://wordpress.org/hello")!)
+            .set(path: "world")
+            .build()
+        XCTAssertEqual(request.url?.absoluteString, "https://wordpress.org/hello/world")
+
+        request = try HTTPRequestBuilder(url: URL(string: "https://wordpress.org/hello")!)
+            .set(path: "/world")
+            .build()
+        XCTAssertEqual(request.url?.absoluteString, "https://wordpress.org/world")
```

Comment:
This behavior, which I read as "when the given `path` is absolute, override the original URL path" is surprising.

Or maybe is just me applying the expectation from `NSString` `appendingPathComponent`:

```
# from the REPL
  2> let s: NSString = "https://foo.test/bar"
s: NSMutableString = "https://foo.test/bar"
  3> s.appendingPathComponent("/baz")
$R0: String = "https:/foo.test/bar/baz"
  4> s.appendingPathComponent("bang")
$R1: String = "https:/foo.test/bar/bang"
  5>
```

I acknowledge that `NSString` explicitly says "appending", though, whereas this API gives no similar clear guardrail on what `path` would do.

Reply:
After a second look, I do agree with you that this behaviour is not very intuitive.

I have updated the API to only allow appending path. See https://github.com/wordpress-mobile/WordPressKit-iOS/pull/657/commits/4bb836093c79c8ee7c897afe42ded12032766f01